app/controllers/medications_controller.rb

```diff
@@ -50,6 +50,9 @@ def edit
   def create
     @medication =
       Medication.new(medication_params.merge(user_id: current_user.id))
+
+    @medication.refill = @medication.refill.nil? ? Time.zone.now.strftime('%d/%m/%Y').to_date +
+      7 * ((@medication.total / @medication.dosage) / @medication.weekly_dosage.length) + ((@medication.total / @medication.dosage) % @medication.weekly_dosage.length) - 1 : @medication.refill
```

Comment:
I would would break apart this logic into a helper function. That should reduce the complexity of this create function and better separate the logic you're adding.

Reply:
I would use `if` and `unless` like Code Climate suggests instead of having multi-line ternary operators. That should also improve readability of the method.

You can run `rubocop` locally and it should throw these warnings.

Reply:
Thanks! I have fixed those issues, now there is no error in codeclimate. However, the total-coverage and diff-coverage checks have failed. I did research a bit but I couldn't pinpoint the error. Would be great if you could look into it!